core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -615,6 +615,12 @@ package object config {
     .stringConf
     .createOptional
 
+  private[spark] val METRICS_REGISTER_STATIC_SOURCES_ENABLED =
+    ConfigBuilder("spark.metrics.register.static.sources.enabled")
```

Comment:
`spark.metrics.register.static.sources.enabled` -> `spark.metrics.static.sources.enabled` because `register` and `enabled` has the same meaning here. `enabled` will be enough.